openhands/controller/agent_controller.py

```diff
@@ -191,7 +192,9 @@ async def start_step_loop(self):
         """The main loop for the agent's step-by-step execution."""
 
         self.log('info', 'Starting step loop...')
-        while should_continue():
+        while True:
+            if self._closed:
+                break
```

Comment:
This is probably the biggest fix 🙈 

These loops were continuing indefinitely even after a session was closed

Reply:
Will the `while True` re-introduce the unable to interrupt issue @tofarr spotted earlier?

Reply:
woof good catch, that was me debugging. Reverted the `while True` part, kept the `self._closed` part